lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -387,19 +467,39 @@ class MetricsEndpointConsumer(Object):
 
     on = MonitoringEvents()
 
-    def __init__(self, charm, name):
+    def __init__(self, charm: CharmBase, relation_name: Optional[str] = None):
```

Comment:
iirc, `Optional[str] = None` is the same as `str = None` for th static inference